Spatial_Engine/Query/IsSelfIntersecting.cs

```diff
@@ -20,19 +20,30 @@
  * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
  */
 
-using BH.Engine.Geometry;
 using BH.oM.Dimensional;
 using BH.oM.Geometry;
 using BH.oM.Quantities.Attributes;
 using BH.oM.Reflection.Attributes;
-using System;
-using System.Collections.Generic;
 using System.ComponentModel;
 
 namespace BH.Engine.Spatial
 {
     public static partial class Query
     {
+        /******************************************/
+        /****            IElement0D            ****/
+        /******************************************/
+
+        [Description("Returns if the one dimensional representation of the IElement0D is closer to itself than the tolerance at any two points. Always false due to zero-dimensionality of an IElement0D.")]
+        [Input("element0D", "The IElement0D to evaluate self intersections from.")]
+        [Input("tolerance", "Minimum distance to be considered intersecting.", typeof(Length))]
+        [Output("A boolean which is true if the IElement0D is self intersecting.")]
```

Comment:
```suggestion
        [Description("Returns if the geometrical representation of an IElement0D is closer to itself than the tolerance at any two points. Always false due to zero-dimensionality of an IElement0D.")]
        [Input("element0D", "The IElement0D to evaluate self intersections from.")]
        [Input("tolerance", "Minimum distance to be considered intersecting.", typeof(Length))]
        [Output("A boolean which is true if the geometrical representation of an IElement0D is self intersecting.")]
```
I don't think the `IElementd0D` becomes 1D at any point here. I think the following sentence makes the description clear enough.